### ISSUE TITLE
Suffix detection broken for private `uk.com` suffix in version 3.4.3

### ISSUE THREAD

**Issue by kevinmarsh** [john-kurkowski/tldextract#288]
I _think_ the trie suffix detection in #285 in version 3.4.3 might have broken looking up `uk.com` private suffix (which is included in the bundled snapshot)
https://github.com/john-kurkowski/tldextract/blob/6f45fed6c56f377e8a9a77ce43c50712281940d8/tldextract/.tld_set_snapshot#L10570

Comparing 3.4.2:
```python
>>> import tldextract
>>> tldextract.__version__
'3.4.2'
>>> extractor = tldextract.TLDExtract(include_psl_private_domains=True)
>>> extractor("foo.uk.com")
ExtractResult(subdomain='', domain='foo', suffix='uk.com')
```

to 3.4.3:
```python
>>> import tldextract
>>> tldextract.__version__
'3.4.3'
>>> extractor = tldextract.TLDExtract(include_psl_private_domains=True)
>>> extractor("foo.uk.com")
ExtractResult(subdomain='foo', domain='uk', suffix='com')
```

you can see that the `uk.com` suffix is no longer recognized but instead thinks `uk` is the domain.

Although weirdly just using the `tldextract.extract` wrapper function in both versions give the exact same (correct) results
```python
>>> import tldextract
>>> tldextract.extract("foo.uk.com", include_psl_private_domains=True)
ExtractResult(subdomain='', domain='foo', suffix='uk.com')
```

**Comment by john-kurkowski** [comment]
I'm looking into this. /cc @elliotwutingfeng

**Comment by john-kurkowski** [comment]
Fixed in 3.4.4. Thanks for the detailed report! That really eased tracking down the bug.
